lib/sync/pipeline.ts

```diff
@@ -141,96 +140,90 @@ export const importCards = async (
 
 	for (const [index, value] of sequence.entries()) {
 		const step = _.castArray(value);
-		await Bluebird.map(
-			step,
-			async (segment, subindex, length) => {
-				const path = ['cards', index];
-				if (length !== 1) {
-					path.push(subindex);
-				}
-
-				let object = {};
-				let finalObject: Partial<Contract> = {};
-				const type = segment.card.type;
-
-				// Check if this is a JSONpatch or a slug-based upsert
-				if ('patch' in segment.card) {
-					// If the patch doesn't update the origin, add it now
-					if (
-						!_.find(segment.card.patch, {
-							path: '/data/origin',
-						})
-					) {
-						if (
-							options.origin &&
-							options.origin.type === 'external-event@1.0.0'
-						) {
-							segment.card.patch.push({
-								op: 'add',
-								path: '/data/origin',
-								value: `${options.origin.slug}@${options.origin.version}`,
-							});
-						}
-					}
-					finalObject = evaluateObject(
-						_.omit(segment.card, ['links']),
-						references,
-					);
-				} else {
-					object = evaluateObject(_.omit(segment.card, ['links']), references);
-					assert.INTERNAL(context, !!object, errors.SyncInvalidTemplate, () => {
-						return `Could not evaluate template in: ${JSON.stringify(
-							segment.card,
-							null,
-							2,
-						)}`;
-					});
+		step.map(async (segment: SequenceItem, subindex: number) => {
```

Comment:
This won't have the desired behaviour, as the internal promise won't be awaited correctly. You'll need to wrap this in a `Promise.map` to get the code to correctly wait for execution, or alternatively use a for loop instead of a map.
```
for (const segment of step) {
}
```

Reply:
@LucianBuzzo Nice catch, missed that as I was just looking at converting `Bluebird.map` to a native alternative. I'll update to a `for const` loop, that's the most legible in my opinion. And unless I'm missing something, there isn't a native `Promise.map` option.

Reply:
Updated to `for const` loop, but want to double-check that it behaves the way I expect. This block is a bit cryptic.